rdt/transformers/categorical.py

```diff
@@ -243,11 +268,17 @@ def transform(self, data):
             numpy.ndarray:
         """
         data = self._prepare_data(data)
-        dummies = pd.get_dummies(data, dummy_na=self.dummy_na)
-        array = dummies.reindex(columns=self.dummies, fill_value=0).values.astype(int)
-        for i, row in enumerate(array):
-            if np.all(row == 0) and self.error_on_unknown:
-                raise ValueError(f'The value {data[i]} was not seen during the fit stage.')
+        size = len(data)
+
+        if size < 2800:
```

Comment:
I'm curious about this boundary: Is it always the same independently on the amount of unique categories involved?